src/markdown_utils.cpp

```diff
@@ -73,25 +79,298 @@ std::string MarkdownToText(const std::string& markdown_str) {
     if (markdown_str.empty()) {
         return "";
     }
-    
+
     // Parse the markdown document
     cmark_parser *parser = cmark_parser_new(CMARK_OPT_DEFAULT);
     cmark_parser_feed(parser, markdown_str.c_str(), markdown_str.length());
     cmark_node *doc = cmark_parser_finish(parser);
-    
+
     // Render as plain text
     char *text = cmark_render_plaintext(doc, CMARK_OPT_DEFAULT, 0);
-    
+
     std::string result(text);
-    
+
     // Clean up
     free(text);
     cmark_node_free(doc);
     cmark_parser_free(parser);
-    
+
+    return result;
+}
+
+// Helper function to extract text content from a lexbor node
+static std::string ExtractText(lxb_dom_node_t* node) {
+    if (node == nullptr) {
+        return "";
+    }
+
+    // Use lexbor's built-in text content function
+    size_t text_len = 0;
+    lxb_char_t* text = lxb_dom_node_text_content(node, &text_len);
+
+    if (text == nullptr || text_len == 0) {
+        return "";
+    }
+
+    std::string result(reinterpret_cast<const char*>(text), text_len);
+
+    // Free the allocated text (lexbor allocates memory that we need to free)
+    lxb_dom_document_destroy_text(lxb_dom_interface_document(node->owner_document), text);
+
+    return result;
+}
+
+// UTF-8 safe trim function (StringUtil::Trim is NOT UTF-8 aware)
+static void TrimUTF8(std::string& str) {
+    // Trim leading ASCII whitespace
+    while (!str.empty() && (str.front() == ' ' || str.front() == '\t' || str.front() == '\n' || str.front() == '\r')) {
+        str.erase(0, 1);
+    }
+    // Trim trailing ASCII whitespace
+    while (!str.empty() && (str.back() == ' ' || str.back() == '\t' || str.back() == '\n' || str.back() == '\r')) {
+        str.pop_back();
+    }
+}
+
+// Helper function to convert HTML entity to actual character
+static std::string DecodeHTMLEntities(const std::string& str) {
+    std::string result = str;
+
+    // Common HTML entities
+    StringUtil::Replace(result, "&lt;", "<");
+    StringUtil::Replace(result, "&gt;", ">");
+    StringUtil::Replace(result, "&amp;", "&");
+    StringUtil::Replace(result, "&quot;", "\"");
+    StringUtil::Replace(result, "&apos;", "'");
+    StringUtil::Replace(result, "&#39;", "'");
```

Comment:
Yeah this is quite garbage. The entity encoding from webbed is much much better 👍